git checkout not building on Ubuntu 18.04

**Description**

A clean git checkout of master does not build. `make dev` hangs endlessly.

**Steps to reproduce**

> git clone https://github.com/google/gvisor.git
> cd gvisor/
> make dev
> make dev
> make[1]: Entering directory '/home/josh/Projects/gvisor'
> make[2]: Entering directory '/home/josh/Projects/gvisor'
> Error: No such container: gvisor-bazel-ef81eefe
> make[3]: Entering directory '/home/josh/Projects/gvisor'
> make[4]: Entering directory '/home/josh/Projects/gvisor/images'
> docker inspect gcr.io/gvisor-presubmit/default_x86_64:9dbe41d72a793a4c >/dev/null 2>&1 || make pull-default || make rebuild-default
> docker tag gcr.io/gvisor-presubmit/default_x86_64:9dbe41d72a793a4c gvisor.dev/images/default
> make[4]: Leaving directory '/home/josh/Projects/gvisor/images'
> docker run -d --rm \
>         --init \
>         --name gvisor-bazel-ef81eefe \
>         --user 0:0  \
>         -v "/home/josh/Projects/gvisor:/home/josh/Projects/gvisor" \
>         --workdir "/home/josh/Projects/gvisor" \
>         --entrypoint "" \
>          -v "/home/josh/.cache/bazel:/home/josh/.cache/bazel" -v "/home/josh/.config/gcloud:/home/josh/.config/gcloud" -v "/tmp:/tmp" -v "/var/run/docker.sock:/var/run/docker.sock" --group-add 130 \
>         gvisor.dev/images/default \
>         sh -c "groupadd --gid 1000 --non-unique josh && \
>                useradd --uid 1000 --non-unique --no-create-home --gid 1000  --groups 130 -d /home/josh josh && \
>                bazel version && \
>                exec tail --pid=\$(bazel info server_pid) -f /dev/null"
> 38dc32e0e602681223c256af6c4ac0463c5fcdf10b0c82255eb0230bf3c60e23

If available, please include detailed reproduction steps.

If the bug requires software that is not publicly available, see if it can be
reproduced with software that is publicly available.

**Environment**

Ubuntu 18.04 x86_64

*   `runsc -v`
*   `docker version` or `docker info` (if available)
docker info
Client:
 Debug Mode: false

Server:
 Containers: 10
  Running: 0
  Paused: 0
  Stopped: 10
 Images: 13
 Server Version: 19.03.6
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version:
 runc version:
 init version:
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.3.0-51-generic
 Operating System: Ubuntu 18.04.4 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 15.46GiB
 Name: josh-ZenBook
 ID: AENT:VI3N:3KMO:DSV7:OYBC:UE3C:S42M:DNNP:ACZ3:NU7T:M35A:5LRT
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
*   `kubectl version` and `kubectl get nodes` (if using Kubernetes)
*   `uname -a`
Linux josh-ZenBook 5.3.0-51-generic #44~18.04.2-Ubuntu SMP Thu Apr 23 14:27:18 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
*   `git describe` (if built from source)
release-20200511.0-281-gba2bf9fc
*   `runsc` debug logs (if available)

@amscanne, I pulled your branch and `make dev` no longer hangs, however I still receive an error that I can't quite figure out:
```
make[1]: Entering directory '/home/josh/Projects/gvisor-group-fix'
make[2]: Entering directory '/home/josh/Projects/gvisor-group-fix'
Loading:
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
INFO: Call stack for the definition of repository 'pydeps' which is a pip_import (rule definition at /home/josh/.cache/bazel/_bazel_josh/2e12320e5d6eb7852b598a49137e6d2e/external/rules_python/python/pip.bzl:51:14):
 - <builtin>
 - /home/josh/Projects/gvisor-group-fix/WORKSPACE:107:1
ERROR: An error occurred during the fetch of repository 'pydeps':
   pip_import failed:  (Timed out)
ERROR: no such package '@pydeps//': pip_import failed:  (Timed out)
INFO: Elapsed time: 600.229s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
tools/bazel.mk:110: recipe for target 'copy' failed
make[2]: *** [copy] Error 1
make[2]: Leaving directory '/home/josh/Projects/gvisor-group-fix'
Makefile:186: recipe for target 'refresh' failed
make[1]: *** [refresh] Error 2
make[1]: Leaving directory '/home/josh/Projects/gvisor-group-fix'
Makefile:178: recipe for target 'dev' failed
make: *** [dev] Error 2
```

--- a/tools/bazel.mk
+++ b/tools/bazel.mk
@@ -21,7 +21,8 @@ BRANCH_NAME := $(shell (git branch --show-current 2>/dev/null || \
 
 # Bazel container configuration (see below).
 USER ?= gvisor
-DOCKER_NAME ?= gvisor-bazel-$(shell readlink -m $(CURDIR) | md5sum | cut -c1-8)
+HASH ?= $(shell readlink -m $(CURDIR) | md5sum | cut -c1-8)
+DOCKER_NAME ?= gvisor-bazel-$(HASH)
 DOCKER_PRIVILEGED ?= --privileged
 BAZEL_CACHE := $(shell readlink -m ~/.cache/bazel/)
 GCLOUD_CONFIG := $(shell readlink -m ~/.config/gcloud/)
@@ -40,6 +41,7 @@ FULL_DOCKER_RUN_OPTIONS += -v "$(DOCKER_SOCKET):$(DOCKER_SOCKET)"
 DOCKER_GROUP := $(shell stat -c '%g' $(DOCKER_SOCKET))
 ifneq ($(GID),$(DOCKER_GROUP))
 USERADD_OPTIONS += --groups $(DOCKER_GROUP)
+GROUPADD_DOCKER += groupadd --gid $(DOCKER_GROUP) --non-unique docker-$(HASH) &&
 FULL_DOCKER_RUN_OPTIONS += --group-add $(DOCKER_GROUP)
 endif
 endif
@@ -71,10 +73,12 @@ bazel-server-start: load-default ## Starts the bazel server.
 		$(FULL_DOCKER_RUN_OPTIONS) \
 		gvisor.dev/images/default \
 		sh -c "groupadd --gid $(GID) --non-unique $(USER) && \
+		       $(GROUPADD_DOCKER) \
 		       useradd --uid $(UID) --non-unique --no-create-home --gid $(GID) $(USERADD_OPTIONS) -d $(HOME) $(USER) && \
 	               bazel version && \
 		       exec tail --pid=\$$(bazel info server_pid) -f /dev/null"
-	@while :; do if docker logs $(DOCKER_NAME) 2>/dev/null | grep "Build label:" >/dev/null; then break; fi; sleep 1; done
+	@while :; do if docker logs $(DOCKER_NAME) 2>/dev/null | grep "Build label:" >/dev/null; then break; fi; \
+		if ! docker ps | grep $(DOCKER_NAME); then exit 1; else sleep 1; fi; done
 .PHONY: bazel-server-start
 
 bazel-shutdown: ## Shuts down a running bazel server.

